media_driver/linux/Xe_M/ddi/media_libva_caps_dg2.cpp

```diff
@@ -532,7 +532,18 @@ VAStatus MediaLibvaCapsDG2::GetPlatformSpecificAttrib(VAProfile profile,
         }
         case VAConfigAttribPredictionDirection:
         {
-            *value = VA_PREDICTION_DIRECTION_PREVIOUS | VA_PREDICTION_DIRECTION_FUTURE | VA_PREDICTION_DIRECTION_BI_NOT_EMPTY;
+            if (!IsHevcSccProfile(profile))
+            {
+                *value = VA_PREDICTION_DIRECTION_PREVIOUS | VA_PREDICTION_DIRECTION_FUTURE | VA_PREDICTION_DIRECTION_BI_NOT_EMPTY;
+            }
+            else
+            {
+                // Here we set
+                // VAConfigAttribPredictionDirection: VA_PREDICTION_DIRECTION_PREVIOUS together with
+                // VAConfigAttribEncMaxRefFrames: L0 != 0, L1 !=0
+                // to indicate SCC only supports I/P/low delay B
+                *value = VA_PREDICTION_DIRECTION_PREVIOUS;
```

Comment:
VA_PREDICTION_DIRECTION_BI_NOT_EMPTY should be set for low-delay B, otherwise user won't know reference list 1 should be set too when constructing reference lists.

@XinfengZhang  @HeJunyan Is my understanding right ?

Reply:
VA_PREDICTION_DIRECTION_BI_NOT_EMPTY is set when L0 only frames should not exist. So for P frame, we need to add a duplicated L1 and construct LDB. But for B frames, which already have L1, this flag has no big effect.

Reply:
How do you know l1==l0 ? for B frame ? @HeJunyan

Reply:
let us align again:
| Supported Type | EncMaxRefFrames Setting | PredictionDirection Setting                                                                                |
| -------------- | ----------------------- | ---------------------------------------------------------------------------------------------------------- |
| I/P            | l0 != 0, l1 = 0         | VA_PREDICTION_DIRECTION_PREVIOUS                                                                           |
| I/P/LDB        | l0 != 0, l1 != 0        | VA_PREDICTION_DIRECTION_PREVIOUS                                                                           |
| I/P/B          | l0 != 0, l1 != 0        | VA_PREDICTION_DIRECTION_PREVIOUS \| VA_PREDICTION_DIRECTION_FUTURE                                         |
| I/LDB/B        | l0 != 0, l1 != 0        | VA_PREDICTION_DIRECTION_PREVIOUS \| VA_PREDICTION_DIRECTION_FUTURE \| VA_PREDICTION_DIRECTION_BI_NOT_EMPTY |
| I/LDB          | l0 != 0, l1 != 0        | VA_PREDICTION_DIRECTION_PREVIOUS \| VA_PREDICTION_DIRECTION_BI_NOT_EMPTY                                   |

Is this make sense?

Reply:
Got it, thanks!

Reply:
The VAEncSliceParameterBufferHEVC has ref_pic_list0 and ref_pic_list1 respectively.  The good way, maybe, not let the ref pictures in L0+L1 surplus 3.

Reply:
Right, thanks.